Super-Linter SHELL_SHFMT errors. Shell scripts should be indented with tabs

SHELL_SHFMT
  2025-10-01 18:08:21 [INFO]   Linting SHELL_SHFMT items...
  Error: -01 18:08:21 [ERROR]   Found errors when linting SHELL_SHFMT. Exit code: 1.
  2025-10-01 18:08:21 [INFO]   Command output for SHELL_SHFMT

https://github.com/SalamLang/Salam/actions/runs/18171230777/job/51726195750

--- a/b.sh
+++ b/b.sh
@@ -20,34 +20,34 @@ OBJ_FILES=""
 
 # ----------- Compilation -----------
 if [[ "$COMPILER" == "tcc" ]]; then
-    while IFS= read -r file; do
-        echo "Compiling $file ..."
-        if ! $COMPILER -c $CFLAGS "$file" -o "${file%.*}.o" 2> compile_error.tmp; then
-            echo "Compilation failed for $file:"
-            cat compile_error.tmp
-            rm -f compile_error.tmp
-            exit 1
-        fi
-        rm -f compile_error.tmp
-        OBJ_FILES="$OBJ_FILES ${file%.*}.o"
-    done < "$C_FILES"
+		while IFS= read -r file; do
+				echo "Compiling $file ..."
+				if ! $COMPILER -c $CFLAGS "$file" -o "${file%.*}.o" 2> compile_error.tmp; then
+						echo "Compilation failed for $file:"
+						cat compile_error.tmp
+						rm -f compile_error.tmp
+						exit 1
+				fi
+				rm -f compile_error.tmp
+				OBJ_FILES="$OBJ_FILES ${file%.*}.o"
+		done < "$C_FILES"
 
-    echo "Linking..."
-    if ! $COMPILER -o "$SALAM_OUTPUT" $OBJ_FILES $LDFLAGS; then
-        echo "Linking failed!"
-        exit 1
-    fi
+		echo "Linking..."
+		if ! $COMPILER -o "$SALAM_OUTPUT" $OBJ_FILES $LDFLAGS; then
+				echo "Linking failed!"
+				exit 1
+		fi
 
 elif [[ "$COMPILER" == "gcc" ]]; then
-    echo "Compiling and linking with GCC..."
-    if ! $COMPILER $CFLAGS @"$C_FILES" -o "$SALAM_OUTPUT" $LDFLAGS; then
-        echo "Compilation or linking failed!"
-        exit 1
-    fi
+		echo "Compiling and linking with GCC..."
+		if ! $COMPILER $CFLAGS @"$C_FILES" -o "$SALAM_OUTPUT" $LDFLAGS; then
+				echo "Compilation or linking failed!"
+				exit 1
+		fi
 
 else
-    echo "Unknown compiler: $COMPILER"
-    exit 1
+		echo "Unknown compiler: $COMPILER"
+		exit 1
 fi
 
 echo "Compilation and linking successful."
@@ -55,31 +55,31 @@ echo "Compilation and linking successful."
 # ----------- Run Program -----------
 echo "Running program..."
 if ! ./"$SALAM_OUTPUT" input.salam; then
-    echo "Program execution failed!"
-    exit 1
+		echo "Program execution failed!"
+		exit 1
 fi
 echo "Program executed successfully."
 
 # ────── Function to beautify JSON ──────
 beautify_json_if_valid() {
-    local file="$1"
-    if [ -f "$file" ]; then
-        if command -v jq >/dev/null 2>&1; then
-            echo "Checking if $file is valid JSON..."
-            if jq empty "$file" >/dev/null 2>&1; then
-                echo "Beautifying $file using jq..."
-                jq . "$file" > "${file}.pretty" && mv "${file}.pretty" "$file"
-                echo "Beautification of $file complete."
-            else
-                echo "$file contains invalid JSON. Skipping beautification."
-            fi
-        else
-            echo "Warning: 'jq' is not installed. Skipping beautification of $file."
-            echo "To install: sudo apt install jq"
-        fi
-    else
-        echo "$file not found. Skipping beautification."
-    fi
+		local file="$1"
+		if [ -f "$file" ]; then
+				if command -v jq >/dev/null 2>&1; then
+						echo "Checking if $file is valid JSON..."
+						if jq empty "$file" >/dev/null 2>&1; then
+								echo "Beautifying $file using jq..."
+								jq . "$file" > "${file}.pretty" && mv "${file}.pretty" "$file"
+								echo "Beautification of $file complete."
+						else
+								echo "$file contains invalid JSON. Skipping beautification."
+						fi
+				else
+						echo "Warning: 'jq' is not installed. Skipping beautification of $file."
+						echo "To install: sudo apt install jq"
+				fi
+		else
+				echo "$file not found. Skipping beautification."
+		fi
 }
 
 # ────── Beautify JSON outputs ──────

--- a/b.sh
+++ b/b.sh
@@ -22,7 +22,7 @@ OBJ_FILES=""
 if [[ "$COMPILER" == "tcc" ]]; then
 	while IFS= read -r file; do
 		echo "Compiling $file ..."
-		if ! $COMPILER -c $CFLAGS "$file" -o "${file%.*}.o" 2> compile_error.tmp; then
+		if ! $COMPILER -c $CFLAGS "$file" -o "${file%.*}.o" 2>compile_error.tmp; then
 			echo "Compilation failed for $file:"
 			cat compile_error.tmp
 			rm -f compile_error.tmp
@@ -30,7 +30,7 @@ if [[ "$COMPILER" == "tcc" ]]; then
 		fi
 		rm -f compile_error.tmp
 		OBJ_FILES="$OBJ_FILES ${file%.*}.o"
-	done < "$C_FILES"
+	done <"$C_FILES"
 
 	echo "Linking..."
 	if ! $COMPILER -o "$SALAM_OUTPUT" $OBJ_FILES $LDFLAGS; then
@@ -68,7 +68,7 @@ beautify_json_if_valid() {
 			echo "Checking if $file is valid JSON..."
 			if jq empty "$file" >/dev/null 2>&1; then
 				echo "Beautifying $file using jq..."
-				jq . "$file" > "${file}.pretty" && mv "${file}.pretty" "$file"
+				jq . "$file" >"${file}.pretty" && mv "${file}.pretty" "$file"
 				echo "Beautification of $file complete."
 			else
 				echo "$file contains invalid JSON. Skipping beautification."

--- a/b.sh
+++ b/b.sh
@@ -20,34 +20,34 @@ OBJ_FILES=""
 
 # ----------- Compilation -----------
 if [[ "$COMPILER" == "tcc" ]]; then
-		while IFS= read -r file; do
-				echo "Compiling $file ..."
-				if ! $COMPILER -c $CFLAGS "$file" -o "${file%.*}.o" 2> compile_error.tmp; then
-						echo "Compilation failed for $file:"
-						cat compile_error.tmp
-						rm -f compile_error.tmp
-						exit 1
-				fi
-				rm -f compile_error.tmp
-				OBJ_FILES="$OBJ_FILES ${file%.*}.o"
-		done < "$C_FILES"
-
-		echo "Linking..."
-		if ! $COMPILER -o "$SALAM_OUTPUT" $OBJ_FILES $LDFLAGS; then
-				echo "Linking failed!"
-				exit 1
+	while IFS= read -r file; do
+		echo "Compiling $file ..."
+		if ! $COMPILER -c $CFLAGS "$file" -o "${file%.*}.o" 2> compile_error.tmp; then
+			echo "Compilation failed for $file:"
+			cat compile_error.tmp
+			rm -f compile_error.tmp
+			exit 1
 		fi
+		rm -f compile_error.tmp
+		OBJ_FILES="$OBJ_FILES ${file%.*}.o"
+	done < "$C_FILES"
+
+	echo "Linking..."
+	if ! $COMPILER -o "$SALAM_OUTPUT" $OBJ_FILES $LDFLAGS; then
+		echo "Linking failed!"
+		exit 1
+	fi
 
 elif [[ "$COMPILER" == "gcc" ]]; then
-		echo "Compiling and linking with GCC..."
-		if ! $COMPILER $CFLAGS @"$C_FILES" -o "$SALAM_OUTPUT" $LDFLAGS; then
-				echo "Compilation or linking failed!"
-				exit 1
-		fi
+	echo "Compiling and linking with GCC..."
+	if ! $COMPILER $CFLAGS @"$C_FILES" -o "$SALAM_OUTPUT" $LDFLAGS; then
+		echo "Compilation or linking failed!"
+		exit 1
+	fi
 
 else
-		echo "Unknown compiler: $COMPILER"
-		exit 1
+	echo "Unknown compiler: $COMPILER"
+	exit 1
 fi
 
 echo "Compilation and linking successful."
@@ -55,31 +55,31 @@ echo "Compilation and linking successful."
 # ----------- Run Program -----------
 echo "Running program..."
 if ! ./"$SALAM_OUTPUT" input.salam; then
-		echo "Program execution failed!"
-		exit 1
+	echo "Program execution failed!"
+	exit 1
 fi
 echo "Program executed successfully."
 
 # ────── Function to beautify JSON ──────
 beautify_json_if_valid() {
-		local file="$1"
-		if [ -f "$file" ]; then
-				if command -v jq >/dev/null 2>&1; then
-						echo "Checking if $file is valid JSON..."
-						if jq empty "$file" >/dev/null 2>&1; then
-								echo "Beautifying $file using jq..."
-								jq . "$file" > "${file}.pretty" && mv "${file}.pretty" "$file"
-								echo "Beautification of $file complete."
-						else
-								echo "$file contains invalid JSON. Skipping beautification."
-						fi
-				else
-						echo "Warning: 'jq' is not installed. Skipping beautification of $file."
-						echo "To install: sudo apt install jq"
-				fi
+	local file="$1"
+	if [ -f "$file" ]; then
+		if command -v jq >/dev/null 2>&1; then
+			echo "Checking if $file is valid JSON..."
+			if jq empty "$file" >/dev/null 2>&1; then
+				echo "Beautifying $file using jq..."
+				jq . "$file" > "${file}.pretty" && mv "${file}.pretty" "$file"
+				echo "Beautification of $file complete."
+			else
+				echo "$file contains invalid JSON. Skipping beautification."
+			fi
 		else
-				echo "$file not found. Skipping beautification."
+			echo "Warning: 'jq' is not installed. Skipping beautification of $file."
+			echo "To install: sudo apt install jq"
 		fi
+	else
+		echo "$file not found. Skipping beautification."
+	fi
 }
 
 # ────── Beautify JSON outputs ──────

--- a/build_and_run.sh
+++ b/build_and_run.sh
@@ -18,17 +18,17 @@ if [ $EXIT_CODE -ne 0 ]; then
 fi
 
 beautify_json_if_valid() {
-local file="$1"
-if [ -f "$file" ]; then
-		if command -v jq >/dev/null 2>&1; then
-				if jq empty "$file" >/dev/null 2>&1; then
-						jq . "$file" > "${file}.pretty" && mv "${file}.pretty" "$file"
-						echo "Beautified $file"
-				else
-						echo "$file is invalid JSON. Skipping."
-				fi
-		fi
-fi
+	local file="$1"
+	if [ -f "$file" ]; then
+			if command -v jq >/dev/null 2>&1; then
+					if jq empty "$file" >/dev/null 2>&1; then
+							jq . "$file" > "${file}.pretty" && mv "${file}.pretty" "$file"
+							echo "Beautified $file"
+					else
+							echo "$file is invalid JSON. Skipping."
+					fi
+			fi
+	fi
 }
 
 beautify_json_if_valid src/tokens.json

--- a/build_and_run.sh
+++ b/build_and_run.sh
@@ -4,8 +4,8 @@ cd build || exit 1
 
 make
 if [ $? -ne 0 ]; then
-		echo "Build failed!"
-		exit 1
+	echo "Build failed!"
+	exit 1
 fi
 
 echo "Running program with Valgrind..."
@@ -13,22 +13,22 @@ valgrind -s --leak-check=full --show-leak-kinds=all --track-origins=yes ./salam 
 
 EXIT_CODE=$?
 if [ $EXIT_CODE -ne 0 ]; then
-		echo "Program failed. Skipping beautification."
-		exit $EXIT_CODE
+	echo "Program failed. Skipping beautification."
+	exit $EXIT_CODE
 fi
 
 beautify_json_if_valid() {
-		local file="$1"
-		if [ -f "$file" ]; then
-				if command -v jq >/dev/null 2>&1; then
-						if jq empty "$file" >/dev/null 2>&1; then
-								jq . "$file" > "${file}.pretty" && mv "${file}.pretty" "$file"
-								echo "Beautified $file"
-						else
-								echo "$file is invalid JSON. Skipping."
-						fi
+local file="$1"
+if [ -f "$file" ]; then
+		if command -v jq >/dev/null 2>&1; then
+				if jq empty "$file" >/dev/null 2>&1; then
+						jq . "$file" > "${file}.pretty" && mv "${file}.pretty" "$file"
+						echo "Beautified $file"
+				else
+						echo "$file is invalid JSON. Skipping."
 				fi
 		fi
+fi
 }
 
 beautify_json_if_valid src/tokens.json

--- a/build_and_run.sh
+++ b/build_and_run.sh
@@ -20,14 +20,14 @@ fi
 beautify_json_if_valid() {
 	local file="$1"
 	if [ -f "$file" ]; then
-			if command -v jq >/dev/null 2>&1; then
-					if jq empty "$file" >/dev/null 2>&1; then
-							jq . "$file" > "${file}.pretty" && mv "${file}.pretty" "$file"
-							echo "Beautified $file"
-					else
-							echo "$file is invalid JSON. Skipping."
-					fi
+		if command -v jq >/dev/null 2>&1; then
+			if jq empty "$file" >/dev/null 2>&1; then
+				jq . "$file" >"${file}.pretty" && mv "${file}.pretty" "$file"
+				echo "Beautified $file"
+			else
+				echo "$file is invalid JSON. Skipping."
 			fi
+		fi
 	fi
 }
 

--- a/build_and_run.sh
+++ b/build_and_run.sh
@@ -4,8 +4,8 @@ cd build || exit 1
 
 make
 if [ $? -ne 0 ]; then
-    echo "Build failed!"
-    exit 1
+		echo "Build failed!"
+		exit 1
 fi
 
 echo "Running program with Valgrind..."
@@ -13,22 +13,22 @@ valgrind -s --leak-check=full --show-leak-kinds=all --track-origins=yes ./salam 
 
 EXIT_CODE=$?
 if [ $EXIT_CODE -ne 0 ]; then
-    echo "Program failed. Skipping beautification."
-    exit $EXIT_CODE
+		echo "Program failed. Skipping beautification."
+		exit $EXIT_CODE
 fi
 
 beautify_json_if_valid() {
-    local file="$1"
-    if [ -f "$file" ]; then
-        if command -v jq >/dev/null 2>&1; then
-            if jq empty "$file" >/dev/null 2>&1; then
-                jq . "$file" > "${file}.pretty" && mv "${file}.pretty" "$file"
-                echo "Beautified $file"
-            else
-                echo "$file is invalid JSON. Skipping."
-            fi
-        fi
-    fi
+		local file="$1"
+		if [ -f "$file" ]; then
+				if command -v jq >/dev/null 2>&1; then
+						if jq empty "$file" >/dev/null 2>&1; then
+								jq . "$file" > "${file}.pretty" && mv "${file}.pretty" "$file"
+								echo "Beautified $file"
+						else
+								echo "$file is invalid JSON. Skipping."
+						fi
+				fi
+		fi
 }
 
 beautify_json_if_valid src/tokens.json

